scripts/label_all_tokens.py

```diff
@@ -0,0 +1,116 @@
+import pickle
+import sys
+from pathlib import Path
+
+from tqdm.auto import tqdm
+from transformers import PreTrainedTokenizer, PreTrainedTokenizerFast
+
+from delphi.eval import token_labelling
+
+
+def tokenize(
+    tokenizer: PreTrainedTokenizer | PreTrainedTokenizerFast, sample_txt: str
+) -> int:
+    # supposedly this can be different than prepending the bos token id
+    return tokenizer.encode(tokenizer.bos_token + sample_txt, return_tensors="pt")[0]
+
+
+# Decode a sentence
+def decode(
+    tokenizer: PreTrainedTokenizer | PreTrainedTokenizerFast, token_ids: list[int]
+) -> str:
+    return tokenizer.decode(token_ids, skip_special_tokens=True)
+
+
+def main():
+    print("\n", " LABEL ALL TOKENS ".center(50, "="), "\n")
+    # Access command-line arguments
+    args = sys.argv[1:]
+    # Directory to save the results
+    SAVE_DIR = Path("src/delphi/eval/")
+
+    # Check if arguments are provided
+    if len(args) == 0:
+        print("No arguments provided.")
+        return
+
+    if len(args) > 1:
+        print("Too many arguments provided.")
+        return
+
+    # Process arguments
+    model_name = args[0]
+
+    print(f"You chose the model: {model_name}\n")
+    print(
+        f"The language model will be loaded from Huggingface and its tokenizer used to do two things:\n\t1) Create a list of all tokens in the tokenizer's vocabulary.\n\t2) Label each token with its part of speech, dependency, and named entity recognition tags.\nThe respective results will be saved to files located at: '{SAVE_DIR}'\n"
+    )
+
+    # ================ (1) =================
+    print("(1) Create a list of all tokens in the tokenizer's vocabulary ...")
+
+    # Load the tokenizer from Huggingface
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
```

Comment:
is this the line that requires sentencepiece and protobuf?

Reply:
yes exactly